map/apply-scheduled-updates-to-preplanned-map-area/src/main/java/com/esri/samples/apply_scheduled_updates_to_preplanned_map_area/ApplyScheduledUpdatesToPreplannedMapAreaSample.java

```diff
@@ -0,0 +1,234 @@
+          ListenableFuture<OfflineMapUpdatesInfo> offlineMapUpdatesInfoFuture = offlineMapSyncTask.checkForUpdatesAsync();
+          offlineMapUpdatesInfoFuture.addDoneListener(() -> {
+            try {
+              // get and check the results
+              OfflineMapUpdatesInfo offlineMapUpdatesInfo = offlineMapUpdatesInfoFuture.get();
+
+              // update UI for available updates
+              if (offlineMapUpdatesInfo.getDownloadAvailability() == OfflineUpdateAvailability.AVAILABLE) {
+                updateAvailableLabel.setText("Updates: AVAILABLE");
+
+                // check and show update size
+                long updateSize = offlineMapUpdatesInfo.getScheduledUpdatesDownloadSize();
+                updateSizeLabel.setText("Update size: " + updateSize + " bytes.");
+
+                // hide the progress indicator
+                progressIndicator.setVisible(false);
+
+                // enable the 'Apply Scheduled Updates' button
+                applyUpdatesButton.setDisable(false);
+                // when the button is clicked, synchronize the mobile map package
+                applyUpdatesButton.setOnAction(e -> {
+
+                  // show progress indicator
+                  progressIndicator.setVisible(true);
+
+                  // create default parameters for the sync task
+                  ListenableFuture<OfflineMapSyncParameters> offlineMapSyncParametersFuture = offlineMapSyncTask.createDefaultOfflineMapSyncParametersAsync();
+                  offlineMapSyncParametersFuture.addDoneListener(() -> {
+                    try {
+                      OfflineMapSyncParameters offlineMapSyncParameters = offlineMapSyncParametersFuture.get();
+
+                      // set the sync direction to none, since we only want to update
+                      offlineMapSyncParameters.setSyncDirection(SyncGeodatabaseParameters.SyncDirection.NONE);
+                      // set the parameters to download all updated for the mobile map packages
+                      offlineMapSyncParameters.setPreplannedScheduledUpdatesOption(PreplannedScheduledUpdatesOption.DOWNLOAD_ALL_UPDATES);
+                      // set the map package to rollback to the old state should the sync job fail
+                      offlineMapSyncParameters.setRollbackOnFailure(true);
+
+                      // create a sync job using the parameters
+                      OfflineMapSyncJob offlineMapSyncJob = offlineMapSyncTask.syncOfflineMap(offlineMapSyncParameters);
+
+                      // start the job and get the results
+                      offlineMapSyncJob.start();
+                      offlineMapSyncJob.addJobDoneListener(() -> {
+                        if (offlineMapSyncJob.getStatus() == Job.Status.SUCCEEDED) {
+                          OfflineMapSyncResult offlineMapSyncResult = offlineMapSyncJob.getResult();
+
+                          // reopen the mobile map package if required
```

Comment:
"if mobile map package reopen is required, close the existing mobile map package and load it again"